packages/itwin/tree-widget/src/components/trees/models-tree/ModelsVisibilityHandler.ts

```diff
@@ -560,10 +578,11 @@ async function createGroupedElementsInfo(imodel: IModelConnection, rulesetId: st
 
   let modelId, categoryId;
   const query = `SELECT Model.Id AS modelId, Category.Id AS categoryId FROM bis.GeometricElement3d WHERE ECInstanceId = ? LIMIT 1`;
-  for await (const modelAndCategoryIds of imodel.query(query, QueryBinder.from([elementId.value]), { rowFormat: QueryRowFormat.UseJsPropertyNames })) {
-    modelId = modelAndCategoryIds.modelId;
-    categoryId = modelAndCategoryIds.categoryId;
-    break;
+  const reader = imodel.createQueryReader(query, QueryBinder.from([elementId.value]), { rowFormat: QueryRowFormat.UseJsPropertyNames });
+  if (await reader.step()) {
+    const row = reader.current.toRow();
+    modelId = row.modelId;
+    categoryId = row.categoryId;
```

Comment:
It should be possible to replace this with `modelId = reader.current.modelId` and `categoryId = row.current.categoryId`